### PR TITLE
Always add gs.trustedHosts to the CSP

### PR DESCRIPTION
**Pull request by juliushaertl**
Improve handling of registering trusted global scale hosts for the CSP allow list and add some config examples.

### CODE REVIEW

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -212,11 +212,7 @@ public function updateCSP() {
 		/**
 		 * Dynamically add CSP for federated editing
 		 */
-		$path = '';
-		try {
-			$path = $container->getServer()->getRequest()->getPathInfo();
-		} catch (\Exception $e) {}
-		if ((strpos($path, '/apps/files/') === 0 || strpos($path, '/s/') === 0) && $container->getServer()->getAppManager()->isEnabledForUser('federation')) {
+		if ($container->getServer()->getAppManager()->isEnabledForUser('federation')) {
```
Comment (juliushaertl) [inline lib/AppInfo/Application.php:215]: Avoiding the files app check as any route could show collabora through the viewer.

**Review by eneiluj — APPROVED** [review]
LGTM code wise

**Pull request merged**